### ISSUE TITLE
`Builtin.set_suite_(documentation|metadata)` raises `AttributeError` instead of `RobotNotRunningError`

### ISSUE THREAD

**Issue by memoselyk** [robotframework/robotframework#2129]
Issue is present in version 2.9.1.
When set_suite_metadata (from BuiltIn library) is called outside a running robotframework context,
the "EXECUTION_CONTEXTS" is not checked to be None and an AttributeError is raised.  

Steps to reproduce:

```python
from robot.libraries.BuiltIn import BuiltIn
BuiltIn().set_suite_metadata("X", "Y")
```

Expected behavior:
RobotNotRunningError exception is raised, just like calling replace_variables from a non-robot-running context. E.g.:
```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Python27\lib\site-packages\robot\libraries\BuiltIn.py", line 1180, in replace_variables
    return self._variables.replace_scalar(text)
  File "C:\Python27\lib\site-packages\robot\libraries\BuiltIn.py", line 79, in _variables
    return self._namespace.variables
  File "C:\Python27\lib\site-packages\robot\libraries\BuiltIn.py", line 71, in _namespace
    return self._context.namespace
  File "C:\Python27\lib\site-packages\robot\libraries\BuiltIn.py", line 66, in _context
    raise RobotNotRunningError('Cannot access execution context')
robot.libraries.BuiltIn.RobotNotRunningError: Cannot access execution context
```

Actual behavior:
This other exception is raised:
```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Python27\lib\site-packages\robot\libraries\BuiltIn.py", line 2872, in set_suite_metadata
    metadata = self._get_namespace(top).suite.metadata
  File "C:\Python27\lib\site-packages\robot\libraries\BuiltIn.py", line 75, in _get_namespace
    return ctx.namespace
AttributeError: 'NoneType' object has no attribute 'namespace'
```

**Comment by pekkaklarck** [comment]
Same problem also with `BuiltIn.set_suite_documentation`. The reason is that these two methods access execution context differently than others and they do not check is the context `None` or not.

**Comment by pekkaklarck** [comment]
Fixed by d57baece9f43501f8c8cef5788ddf892aa1db4ea.
